java/driver/flight-sql/src/main/java/org/apache/arrow/adbc/driver/flightsql/FlightSqlConnection.java

```diff
@@ -62,7 +74,51 @@ public class FlightSqlConnection implements AdbcConnection {
                     throw new RuntimeException(e);
                   }
                 })
-            .build(location -> FlightClient.builder(allocator, location).build());
+            .build(
+                loc -> {
+                  FlightClient flightClient =
+                      FlightClient.builder(allocator, loc)
+                          .intercept(
+                              new FlightClientMiddleware.Factory() {
```

Comment:
nit: could we pull this out into a static inner class instead of having it inline here?

Reply:
That would make Factory a doubly nested inner class, probably best if I split it into an entirely separate file.